frontend/src/components/admin/ApprovedLanguagesTable.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  Tfoot,
+  Button,
+  Slider,
+  SliderTrack,
+  SliderFilledTrack,
+  SliderThumb,
+} from "@chakra-ui/react";
+import { ApprovedLanguagesMap, generateSortFn } from "../../utils/Utils";
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+
+type ApprovedLanguagesTableProps = {
+  approvedLanguagesTranslation: ApprovedLanguagesMap | undefined;
+  approvedLanguagesReview: ApprovedLanguagesMap | undefined;
+};
+
+interface ApprovedLanguageFieldSortDict {
+  [field: string]: {
+    sortFn: (a: ApprovedLanguage, b: ApprovedLanguage) => number;
+    isAscending: boolean;
+    setIsAscending: Dispatch<SetStateAction<boolean>>;
+  };
+}
+
+type ApprovedLanguage = {
+  language: string;
+  level: number;
+  role: string;
+};
+
+const ApprovedLanguagesTable = ({
+  approvedLanguagesTranslation,
+  approvedLanguagesReview,
+}: ApprovedLanguagesTableProps) => {
+  const [approvedLanguages, setApprovedLanguages] = useState<
+    ApprovedLanguage[]
+  >([]);
+
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+
+  const fieldSortDict: ApprovedLanguageFieldSortDict = {
+    language: {
+      sortFn: generateSortFn<ApprovedLanguage>("language", isAscendingLanguage),
+      isAscending: isAscendingLanguage,
+      setIsAscending: setIsAscendingLanguage,
+    },
+    role: {
+      sortFn: generateSortFn<ApprovedLanguage>("role", isAscendingRole),
+      isAscending: isAscendingRole,
+      setIsAscending: setIsAscendingRole,
+    },
+  };
+
+  const sortApprovedLanguages = (field: string) => {
+    const newApprovedLanguages = [...approvedLanguages];
+    const { sortFn, isAscending, setIsAscending } = fieldSortDict[field];
+    setIsAscending(!isAscending);
+    newApprovedLanguages.sort(sortFn);
+    console.log(newApprovedLanguages);
+    setApprovedLanguages(newApprovedLanguages);
+  };
+
+  useEffect(() => {
+    if (approvedLanguagesTranslation && approvedLanguagesReview) {
+      const translationLanguages = Object.entries(
+        approvedLanguagesTranslation,
+      ).map(([language, level]) => {
+        return { language, level, role: "Translator" };
+      });
+      const reviewLanguages = Object.entries(approvedLanguagesTranslation).map(
```

Comment:
little bug: `reviewLanguages` is using `approvedLanguagesTranslation`. Eg carl sagan's table (he has no approved review languages)
![image](https://user-images.githubusercontent.com/15113249/143329816-9b26c80c-f339-48c8-9e86-2bc85535f1c6.png)

Reply:
shoot mb, will fix that